org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend

```diff
@@ -0,0 +1,171 @@
+					type = 'namespace'
+					children = newArrayList
+					children += objects.filter(MacroCollection).map[generate(null)]
+					children += objects.filter(Component).map[generate(null)]
+				]
+			]
+		]
+		return resultTree
+	}
+
+	private def dispatch SerializableStepTreeNode generate(MacroCollection macroCollection, NamedElement context) {
+		return new SerializableStepTreeNode => [
+			displayName = macroCollection.name
+			type = 'macroCollection'
+			children = newArrayList
+			children += macroCollection.macros.map[generate(macroCollection)]
+		]
+	}
+
+	private def dispatch SerializableStepTreeNode generate(Macro macro, NamedElement context) {
+		return new SerializableStepTreeNode => [
+			displayName = macro.template.contents.map[getText(macro)].join(" ")
+			type = 'macro'
+			children = emptyList // will never hold any children
+		]
+
+	}
+
+	private def dispatch SerializableStepTreeNode generate(Component component, NamedElement context) {
+		return new SerializableStepTreeNode => [
+			displayName = component.name
+			type = 'component'
+			children = newArrayList
+			val allInteractions = modelUtil.getComponentInteractionTypes(component)
+			children += allInteractions.map[generate(component)]
+			children += modelUtil.getComponentElements(component).map[generate(component)]
+		]
+
+	}
+
+	private def dispatch SerializableStepTreeNode generate(InteractionType interaction, NamedElement context) {
+		return new SerializableStepTreeNode => [
+			displayName = interaction.template.contents.map[getText(context)].join(' ')
+			type = 'interaction'
+			children = emptyList // will never hold children
+		]
+	}
+
+	private def dispatch SerializableStepTreeNode generate(ComponentElement element, NamedElement context) {
+		return new SerializableStepTreeNode => [
+			displayName = element.name
+			type = 'element'
+			children = newArrayList
+			children += modelUtil.getComponentElementInteractionTypes(element).map[generate(element)]
+		]
+	}
+
+	private def dispatch String getText(TemplateText element, NamedElement context) {
+		element.value
+	}
+
+	private def dispatch String getText(TemplateVariable element, NamedElement context) {
+		if (element.name == "element") {
+			if (context instanceof ComponentElement) {
+				'''<«context.name»>''' // reference to a component element of the context
+			} else {
+				'''<«element.name»>''' // reference to a component element
+			}
+		} else {
+			'''"«element.name»"''' // regular parameter
+		}
+	}
+
+	private def Map<String, List<NamedElement>> getNamespaceToNamedElementsMap(Iterable<NamedElement> namedElements, Function1<NamedElement, String> generateKey) {
+		val resolvedNamedElements = namedElements.map[EcoreUtil2.resolve(it, resourceDescriptionsProvider.indexResourceSet)] //
+		.filter(NamedElement)
+		val namespaceMacroCollectionsMap = resolvedNamedElements.groupBy [ namedElement |
+			val key = generateKey.apply(namedElement)
+			if (key.nullOrEmpty) {
+				return "./."
```

Comment:
I don't really get this literal, where is this coming from / what's the meaning?

Reply:
it's the default namespace. couldn't think of something better.